internal/common/environment.go

```diff
@@ -23,3 +28,16 @@ func EnvOrDefault(envName string, defVal string) string {
 func GetOperatorVersion() string {
 	return EnvOrDefault(OperatorVersionKey, defaultOperatorVersion)
 }
+
+func IsSingleNodeMode(topologyMode osconfv1.TopologyMode) bool {
+	return topologyMode == osconfv1.SingleReplicaTopologyMode
+}
+
+func GetInfrastructureTopology(c client.Client) (osconfv1.TopologyMode, error) {
+	infraConfig := &osconfv1.Infrastructure{}
+	if err := c.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
+		return osconfv1.SingleReplicaTopologyMode, err
```

Comment:
Is it OK to return `SingleReplicaTopologyMode` in case of any error? It seems suboptimal, if for example there is a network error, we assume single node topology.

Maybe on an error we should restart the SSP operator to try again?

Reply:
I was checking the error value and restarting, but you are right, this is confusing. Changed to ""

Reply:
You are right, I missed it.